### PR TITLE
Feat: add  annotations and labels to gateway.cue

### PR DESCRIPTION
**Pull request by YanRyan**
### Description of your changes

add  annotations and labels to gateway.cue

### CODE REVIEW

**Comment by StevenLeiZhang** [comment]
Please fix PR Title Checker

File: charts/vela-core/templates/defwithtemplate/gateway.yaml [hunk 1]
```diff
@@ -125,6 +133,12 @@ spec:
 
         	// +usage=Specify a pathType for the ingress rules, defaults to "ImplementationSpecific"
         	pathType: *"ImplementationSpecific" | "Prefix" | "Exact"
+
```
Comment (StevenLeiZhang) [inline charts/vela-core/templates/defwithtemplate/gateway.yaml:136]: please refer to https://kubevela.net/zh/docs/contributor/cli-ref-doc, and update official doc

File: charts/vela-core/templates/defwithtemplate/gateway.yaml [hunk 2]
```diff
@@ -125,6 +133,12 @@ spec:
...
+        	// +usage=Specify the annotations to be added to the gateway
```
Comment (StevenLeiZhang) [inline charts/vela-core/templates/defwithtemplate/gateway.yaml:137]: I think "ingress" is better than "gateway" in the comment.

File: charts/vela-core/templates/defwithtemplate/gateway.yaml [hunk 3]
```diff
@@ -125,6 +133,12 @@ spec:
...
+        	annotations?: [string]: string | null
+
+        	// +usage=Specify the labels to be added to the gateway
+        	labels?: [string]: string | null
```
Comment (StevenLeiZhang) [inline charts/vela-core/templates/defwithtemplate/gateway.yaml:141]: what does null mean?

File: charts/vela-core/templates/defwithtemplate/gateway.yaml [hunk 4]
```diff
@@ -125,6 +133,12 @@ spec:
 
         	// +usage=Specify a pathType for the ingress rules, defaults to "ImplementationSpecific"
         	pathType: *"ImplementationSpecific" | "Prefix" | "Exact"
+
+        	// +usage=Specify the annotations to be added to the gateway
+        	annotations?: [string]: string | null
```
Comment (StevenLeiZhang) [inline charts/vela-core/templates/defwithtemplate/gateway.yaml:138]: since annotations and labels are array in K8s, why the parameter type is not array?

**Comment by StevenLeiZhang** [comment]
please fix DCO

**Review by FogDong — COMMENTED** [review]
Please sign the DCO.

File: charts/vela-core/templates/defwithtemplate/gateway.yaml (hunk 2 above)
Comment (YanRyan) [reply charts/vela-core/templates/defwithtemplate/gateway.yaml:137]: I will modify it

**Comment by YanRyan** [comment]
fix this issue  “https://github.com/kubevela/kubevela/issues/6396 ”

**Pull request closed without merging**